### PR TITLE
Fix multiple clicks on add to cart freezes page

### LINKED ISSUE #9634: [BOOM-5683] Double click on "Add to cart" on the product page FO freezes the page
> This issue has been migrated from the Forge. Read the [original ticket here](http://forge.prestashop.com/browse/BOOM-5683).

- _**Reporter:**_ Elitius
- _**Created at:**_ Wed, 6 Jun 2018 01:42:57 +0200

----
#### Describe the bug

If you double click on "add to cart" on a product page in FO, the confimation modal is displayed but cannot be closed if you click on the cross or on "continue shipping", but you can proceed to checkout.

#### Expected behavior

I should be able to close the modal

#### Steps to Reproduce

1. Go on a product page in FO
2. Double click on "add to cart"
3. Try to close the confirmation modal, you can't

**Screenshots**

https://drive.google.com/file/d/1ZMlOxzJDPdXoRdnQEgEGSUm-I_irCj_t/view?usp=sharing

#### Additional information

* PrestaShop version: 1.7.7.x and before
* PHP version: N/A

### LINKED ISSUE #15483: FO will be greyed out if you click real fast on add to cart button
**Describe the bug**
If you click like a crazy person on "add to cart" button it opens the modal multiple times

**To Reproduce**
1. Go to a product page on FO
2. Click on 'add to cart' button multiple times, very quick
3. See all the modal open

**Additional information**
PrestaShop version: 1.7.6.1

### PR DESCRIPTION
**Pull request by rdy4ever**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | As stated in #9634 and  #15483, if a user clicks on "add to cart" multiple times (and fast enough), the modal opens and closes multiple times, and then it "freezes", meaning you can no longer close it by clicking on "Continue shopping" or the close button. This also creates the undesired effect of adding to cart more products then intended.
| Type?         | bug fix
| Category?     | FO
| BC breaks?    | Does it break backward compatibility? no
| Deprecations? | Does it deprecate an existing feature?no
| Fixed ticket? | Fixes #9634 Fixes #15483
| How to test?  | See before/after screen rec: https://drive.google.com/open?id=1PMgk5geCcQj0LGaTB-VYCFbQAkos4MAH

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/15760)

### CODE REVIEW

File: themes/_core/js/cart.js [hunk 1]
```diff
@@ -79,9 +79,9 @@ $(document).ready(() => {
     '[data-button-action="add-to-cart"]',
     (event) => {
       event.preventDefault();
-      if ($('#quantity_wanted').val() > $('[data-stock]').data('stock') && $('[data-allow-oosp]').data('allow-oosp').length === 0) {
-          $('[data-button-action="add-to-cart"]').attr('disabled', 'disabled');
-      } else {
+      if ($('#quantity_wanted').val() <= $('[data-stock]').data('stock') && $('[data-allow-oosp]').data('allow-oosp').length !== 0) {
+        $('[data-button-action="add-to-cart"]').attr('disabled', 'disabled');
```
Comment (PierreRambaud) [inline themes/_core/js/cart.js]:
```suggestion
        $('[data-button-action="add-to-cart"]').prop('disabled', true);
```

File: themes/_core/js/cart.js [hunk 2]
```diff
@@ -126,6 +126,7 @@ $(document).ready(() => {
           });
         }).fail((resp) => {
           prestashop.emit('handleError', {eventType: 'addProductToCart', resp: resp});
+          $('[data-button-action="add-to-cart"]').removeAttr('disabled');
```
Comment (PierreRambaud) [inline themes/_core/js/cart.js]:
```suggestion
          $('[data-button-action="add-to-cart"]').removeProp('disabled');
```

**Comment by PierreRambaud** [comment]
Could you please squash your commit and update your develop branch when you create a branch, it generates unnecessary commits history :sweat_smile:

File: themes/_core/js/cart.js (hunk 1 above)
Comment (rdy4ever) [reply themes/_core/js/cart.js]: Sure, done

**Comment by rdy4ever** [comment]
> Could you please squash your commit and update your develop branch when you create a branch, it generates unnecessary commits history 😅

I'm a bit ashamed by this... I'm on GitHub Desktop on Windows and have no ideea how to squash :))

**Comment by PierreRambaud** [comment]
@rdy4ever I only know how to do it in CLI, I'm not a UI user :sweat_smile: 
I can do it for you if you want.

**Comment by rdy4ever** [comment]
> @rdy4ever I only know how to do it in CLI, I'm not a UI user 😅
> I can do it for you if you want.

Sure, go ahead

**Comment by rdy4ever** [comment]
Seems `data-stock` and `data-allow-oos` are not available on quick view and products without stock. I need to check things out, for now this is not reliable.

**Comment by rdy4ever** [comment]
@PierreRambaud 
Just realised we don't need the previous condition at all, because we disable the button anyway... duh :))
Terribly sorry for reformatting the whole function (I know this makes it harder to review the diff), but, with the condition out of the way, the indenting was plain wrong.
Could you please review and squash my many commits as possible? Thanks!

**Comment by PierreRambaud** [comment]
Hi @rdy4ever 
Is this pull request still needed?

**Comment by rdy4ever** [comment]
Well, it's at least outdated. I'll close it and maybe redoit sometime.

**Pull request closed without merging**